### PR TITLE
Import-Export States to your DevTools

### PR DESCRIPTION
**Pull request by pabloalonsos**
I've put together a small addition to the code that would allow us to import and export action logs.

* `Export` button added to top bar.
* Copy the current state from the `textarea` and paste it on another session running the app.

This feature (or something similar) would be awesome to use for QA purposes among other things.

### CODE REVIEW

**Comment by SimenB** [comment]
Huge :+1: Anything keeping this back? @gaearon seemed keen on it in #117

**Comment by gaearon** [comment]
Only that we're splitting `LogMonitor` into a separate project: https://github.com/gaearon/redux-devtools/pull/132. Once that's done, I'll definitely take a look. Sorry it's taking long!

**Comment by SimenB** [comment]
Haha, I love how you define this as a long wait, when it's been a week. I was just wondering if it had fallen through the cracks :smile: No problem there though, apparently. Keep up the great work!

Btw (and sorta related), is there any possibility of getting a sync between browsers/sessions? So you can have different browsers up at the same time all getting the same data automatically. I sorta have it now using browsersync, but that syncs clicks and inputs, and not actual state, which would be nicer, IMO. If it's available, or discussed in another issue, apologies.

**Comment by gaearon** [comment]
>Btw (and sorta related), is there any possibility of getting a sync between browsers/sessions?

You can implement as a store enhancer for regular Redux—it will work with Redux DevTools just fine. You can use `persistState` as inspiration (though it does a different thing).

**Comment by gaearon** [comment]
@pabloalonsos 

Can I ask you to tweak this to:

* Have separate Import and Export buttons on the toolbar (never mind the design; we'll fix it later)
* Remove the notion of "export mode"
* Instead of a text area, show JSON in `window.prompt` for both import and export

?

File: src/devTools.js [hunk 1]
```diff
@@ -120,6 +123,25 @@ function liftReducer(reducer, initialState) {
...
+    case ActionTypes.EXPORT:
+      monitorState = Object.assign(
+        {},
+        monitorState,
+        { exportMode: !monitorState.exportMode }
+      );
+      break;
+    case ActionTypes.IMPORT:
+      (
+        {
```
Comment (gaearon) [inline src/devTools.js]: Please remove extra indentation and line break here.
This is fine:

```js
({
  stagedActions,
  ...
} = liftedAction.newState);
```

Let's also rename `newState` to `nextLiftedState`.

**Comment by pabloalonsos** [comment]
sounds good! I'll try to make the changes today :+1:

**Comment by pabloalonsos** [comment]
There you go. Let me know if all is good and I can squash those commits into one.

There are a couple of "issues" (not really issues but worth bringing up) I ran into:

* By using `window.prompt` instead of the textarea component, ESLint is throwing a couple of warnings. I don't mind finding a different approach if we want to get rid of those warnings without modifying the lint rules (using a custom modal component with a textarea or something like that).
* ESLint was also complaining about having `import` as an action (because of the reserved word) so I switched it to `importState`. I'm pretty bad at picking names for attributes and variables so feel free to suggest alternatives if those are not the best names for this.

Thanks guys!

File: src/devTools.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ const ActionTypes = {
   RESET: 'RESET',
   ROLLBACK: 'ROLLBACK',
   COMMIT: 'COMMIT',
+  IMPORT: 'IMPORT',
```
Comment (gaearon) [inline src/devTools.js]: let's call it `IMPORT_STATE` yup

File: src/react/LogMonitor.js [hunk 3]
```diff
@@ -114,6 +114,21 @@ export default class LogMonitor extends Component {
     this.props.commit();
   }
 
+  handleExport() {
+    window.prompt('Current state\'s schema', JSON.stringify(this.props.store.getState()));
```
Comment (gaearon) [inline src/react/LogMonitor.js]:
```js
/* eslint-disable no-alert */
```

in this file should help

**Comment by pabloalonsos** [comment]
Updated :+1:

**Comment by gaearon** [comment]
One more thing: can you please add some tests?

**Comment by pabloalonsos** [comment]
Busy couple of days at work. I'll try and add the tests before the end of the week!

**Comment by gaearon** [comment]
Sure, thanks! Sorry I'm a bit stuck here too—on a vacation.

**Comment by pabloalonsos** [comment]
Test added! Let me know if anything is missing

**Comment by gaearon** [comment]
I'm bringing this in via b4da7aa in vNext branch.
Closing this PR as is because it won't merge cleanly there.
Thanks!

**Pull request closed without merging**